core-bundle/src/Command/MigrateCommand.php

```diff
@@ -285,17 +285,16 @@ static function ($hash) use ($commandsByHash) {
     private function getCommandHashes(array $commands, bool $withDrops): array
     {
         if (!$withDrops) {
-            unset($commands['ALTER_DROP']);
-
-            foreach ($commands as $category => $commandsByHash) {
-                foreach ($commandsByHash as $hash => $command) {
-                    if ('DROP' === $category && false === strpos($command, 'DROP INDEX')) {
-                        unset($commands[$category][$hash]);
-                    }
+            foreach ($commands as $hash => $command) {
+                if (
+                    (0 === strncmp($command, 'DROP ', 5) && 0 !== strncmp($command, 'DROP INDEX', 10))
```

Comment:
I think the first condition already matches `DROP INDEX`, doesn't it? Then the second one could be omitted.

Reply:
`DROP INDEX` is not considered a “delete” query because it doesn’t remove any data. The second check uses `!==` :)

Reply:
My bad. 🙈